test/e2e/tests/security-provider.spec.js

```diff
@@ -0,0 +1,255 @@
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
+
+        testApp = windowHandles[1];
+
+        await driver.clickElement('#signTypedData');
+
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        const warningHeader = await driver.isElementPresent({
+          text: 'Request may not be safe',
+          tag: 'h5',
+        });
+        assert.equal(warningHeader, true);
+      },
+    );
+  });
+
+  it('Should return not malicious response', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            transactionSecurityCheckEnabled: true,
+          })
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+        testSpecificMock: async (mockServer) =>
+          await mockSecurityProviderDetection(mockServer, 'notMalicious'),
+        dapp: true,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
+
+        testApp = windowHandles[1];
+
+        await driver.clickElement('#siwe');
+
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        const warningHeader = await driver.isElementPresent({
+          text: 'Request may not be safe',
+          tag: 'h5',
+        });
+        assert.equal(warningHeader, false);
+      },
+    );
+  });
+
+  it('Should return request not verified response', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            transactionSecurityCheckEnabled: true,
+          })
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+        testSpecificMock: async (mockServer) =>
+          await mockSecurityProviderDetection(mockServer, 'requestNotVerified'),
+        dapp: true,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
```

Comment:
same here, I think we can remove this line as it's not needed anymore

Reply:
Done